packages/kepler/src/KeplerSlice.ts

```diff
@@ -207,74 +198,105 @@ export function createKeplerSlice<
             ) as Middleware;
             middlewares.push(logger);
           }
-          const dispatchWithMiddleware = (mapId: string, action: KeplerAction) => {
-            const wrapDispatch = (a: Action) => dispatch(mapId, a);
-            wrapDispatch.mapId = mapId;        
-            const middlewareAPI = {
-              getState: get,
-              dispatch: wrapDispatch,
-            };
-            // @ts-ignore
-            applyMiddleware(middlewareAPI, middlewares)(action);
+
+          const storeDispatch = (action: Action) => {
+            set((state: KeplerSliceState<PC>) => ({
+              ...state,
+              kepler: {
+                ...state.kepler,
+                map: keplerReducer(state.kepler.map, action),
+              },
+            }));
+
+            // Call onAction if it's defined
+            get().kepler.onAction?.(mapId, action);
```

Comment:
@ilyabo this will no longer work. Which store is calling kepler.onAction? and what's this for?

Reply:
so it would be called if a client adds an `onAction` implementation to `.kepler` when setting up the store

Reply:
because `sotreDispatch` no longer takes `mapId` as arguments. Would need to get `mapId` from `action.payload.meta._id_`